youtube_dl/extractor/generic.py

```diff
@@ -2009,13 +2009,14 @@ def _extract_rss(self, url, video_id, doc):
 
         entries = []
         for it in doc.findall('./channel/item'):
-            next_url = xpath_text(it, 'link', fatal=False)
-            if not next_url:
-                enclosure_nodes = it.findall('./enclosure')
-                for e in enclosure_nodes:
-                    next_url = e.attrib.get('url')
-                    if next_url:
-                        break
+            enclosure_nodes = it.findall('./enclosure')
+            for e in enclosure_nodes:
+                next_url = e.attrib.get('url')
+                if next_url:
+                    break
+
+            if not enclosure_nodes:
```

Comment:
This does not mean `next_url` is obtained.

Reply:
The check was changed and an assignment was added, so that the error can't occur anymore.

Reply:
Nothing changed here.

Reply:
Is there a change needed here? If `not enclosure_nodes` is `True` it is expected that `next_url` wasn't obtained yet.

Reply:
So what? If `not enclosure_nodes` is `False` it does not mean `next_url` was obtained.

Reply:
After some thought, I think I get what you are saying. If the rss feed has `enclosure_nodes`, but the url is empty it skips even though it shouldn't. This is fixed now.